Frameworks/UIKit/NSTextContainer.mm

```diff
@@ -62,38 +78,120 @@ - (NSArray*)exclusionPaths {
     return _exclusionPaths;
 }
 
+// Finds the first viable region for the proposed rect given the exclusion areas based upon x positions
+// Assumes writing direction is left to right, exclusionLines is sorted and preprocessed for y positions, and all inputs are valid
```

Comment:
This would not be resolved with CGD2D changes, it would need to be a separate task.